src/main/java/org/jenkinsci/plugins/reverse_proxy_auth/auth/DefaultReverseProxyAuthenticator.java

```diff
@@ -49,22 +51,31 @@ public void setMessageSource(@Nonnull MessageSource messageSource) {
 	public void afterPropertiesSet() throws Exception {
 	}
 
-	public ReverseProxyUserDetails authenticate(String username, String password) {
+	public ReverseProxyUserDetails authenticate(String username, String password) throws DataAccessException {
 
 		LOGGER.log(Level.INFO, "DefaultReverseProxyAuthenticator::authenticate ==> {0} to {1}", new Object[]{this.username, authorities});
 
-		ReverseProxyUserDetails userDetails;
+
 		Authentication auth = SecurityContextHolder.getContext().getAuthentication();
-		if (auth != null) {
-			userDetails = (ReverseProxyUserDetails) auth.getPrincipal();
-		} else {
+		if (auth == null) {
 			//do not use the username from the parameters' list.
 			auth = new UsernamePasswordAuthenticationToken(this.username, "", authorities);
 			SecurityContextHolder.getContext().setAuthentication(auth);
-
-			userDetails = (ReverseProxyUserDetails) auth.getPrincipal();
```

Comment:
2 lines above it passes `this.username` String as principal. And then it casts the object to `ReverseProxyUserDetails`... 🤷‍♂️